addons/web/models/ir_http.py

```diff
@@ -62,6 +62,8 @@ def session_info(self):
                 # current_company should be default_company
                 "user_companies": {'current_company': (user.company_id.id, user.company_id.name), 'allowed_companies': [(comp.id, comp.name) for comp in user.company_ids]},
                 "currencies": self.get_currencies(),
+                "company_currency_id": request.env['res.company'].browse(session_info['company_id']).currency_id.id if session_info['company_id'] else None,
```

Comment:
`"company_id": user.company_id.currency_id.id if request.session.uid else None`
To avoid a re-browse.